.github/workflows/ci-testing-deploy.yml

```diff
@@ -276,6 +276,18 @@ jobs:
       fail-fast: false
     name: "[build] docker images"
     steps:
+      - name: Remove unused software
+        run: |
```

Comment:
Q: This looks very heuristic. Hod did you decide on which folders to remove? HOw do you know they will have no side effects? e.g. we use CodeQL, does this folder removal might affect the corresponding action?

Why not simply the gh-action you refer https://github.com/easimon/maximize-build-space?

Reply:
These folders have been discovered by the community. I've known about some of them since a while. And it's a popular way to gain more space.

Looks like the action will be closed and I've seen other repositories that apply this exact solution. This should be more than sufficient for us.